lib/engine/step/buy_sell_par_shares.rb

```diff
@@ -239,15 +243,18 @@ def process_buy_company(action)
         price = action.price
         owner = company.owner
 
-        @game.game_error("Cannot buy #{company.name} from #{owner.name}") unless owner.player?
+        @game.game_error("Cannot buy #{company.name} from #{owner.name}") if owner&.corporation?
 
         company.owner = entity
-        owner.companies.delete(company)
+        owner&.companies&.delete(company)
 
         entity.companies << company
-        entity.spend(price, owner)
+        entity.spend(price, owner.nil? ? @game.bank : owner)
         @current_actions << action
-        @log << "-- #{entity.name} buys #{company.name} from #{owner.name} for #{@game.format_currency(price)}"
+        @log << "#{owner.nil? ? '' : '-- '}"\
+                "#{entity.name} buys #{company.name} from "\
+                "#{owner.nil? ? 'the market' : owner.name} for "\
```

Comment:
same here inverse so you don't have to call explicit nil

Reply:
Fixed